### DISCUSSION TITLE
Referencing a custom plugin in XML

### DISCUSSION THREAD

**Discussion by buddhasystem** [mitsuba-renderer/mitsuba3#1245] [Q&A] [answered]
I decided to try and follow the exercise of creating my BSDF plugin as it is described in Mitsuba documentation. Once I register the plugin, I can't find a way to reference it in the XML definition of a scene. Is that possible? If not, would it not be a desirable feature?

**Comment by merlinND** [comment, marked as answer]
Hello @buddhasystem,

Once the plugin has been registered, you should be able to use `<bsdf type="your_plugin_name">...</bsdf>` (XML) or `{ "type": "your_plugin_name", ... }` (Python `load_dict`).

https://mitsuba.readthedocs.io/en/stable/src/others/custom_plugin.html#Plugin-instantiation

**Comment by buddhasystem** [comment]
Thank you. It did work. One of the variations of the example (containing "tint") didn't really work, but I got another one, from Mitsuba 2, to work. Will be investigating. But it's nice to see that I can modify the reflectance in the Python code and it shows:
![image](https://github.com/user-attachments/assets/dddd66b2-7fb7-416a-b72f-c52b50864942)

**Reply by merlinND** [reply]
@buddhasystem I'll close this discussion as resolved, but feel free to reopen it if needed.
